torch/_dynamo/variables/misc.py

```diff
@@ -1059,9 +1069,14 @@ class ConstantLikeVariable(VariableTracker):
 
     _error_prefix = "ConstantLikeVariable"
     try:
-        from numpy import floating as np_floating
+        from numpy import (
+            dtype as np_dtype,
+            floating as np_floating,
+            generic as np_generic,
+        )
     except ImportError:
         np_floating = type("invalid_type", (), {})
+        np_dtype = type("invalid_type", (), {})
```

Comment:
Why are these in the parent class?

Reply:
Ugh, I see the pattern below...

Reply:
If there were going to be a few more of these `ConstantLike` types coming from numpy I would factor these into an intermediate base, but I don't think we will need many (any?) more. 

LMK if you want me to factor it.

Reply:
nah, it's alright really